internal/dag/policy.go

```diff
@@ -78,7 +83,7 @@ func headersPolicy(policy *projcontour.HeadersPolicy, allowHostRewrite bool) (*H
 		}
 		if key == "Host" {
 			if !allowHostRewrite {
-				return nil, fmt.Errorf("rewriting %q header is not supported", key)
+				return nil, fmt.Errorf("rewriting %q header is only supported on a Spec.Route", key)
```

Comment:
This error msg would be misleading in the case where a user tried to configure a host rewrite on a route's response headers.

I would probably return a more generic error from this function (something closer to what it was before), then augment the error message at the call sites to add more context about why it wasn't allowed.

Reply:
Last push rewrites the error messages so they are more specific.